src/frontend/src/components/form.js

```diff
@@ -186,11 +194,17 @@ class Form extends React.PureComponent<Props> {
             filterOptions={destinationFilterOptions}
             options={locationsWithLabels}
             optionHeight={SELECT_OPTION_HEIGHT}
-            onChange={this.selectDestination}
+            onChange={selectDestination}
             placeholder={t("Geocoding.StartPlaceholder")}
             style={SELECT_STYLE}
             wrapperStyle={SELECT_WRAPPER_STYLE}
-            value={destination.value}
+            value={{
+              ...destination,
+              label:
+                t(`TripPurpose.${destination.purpose ? destination.purpose : destination.value}`) +
```

Comment:
I think this ternary can be avoided by simply adding `purpose: d.purpose` on [line 148](https://github.com/azavea/echo-locator/pull/536/files#diff-bbddc51addecf1196a0906c70a4a81c4686a5584d94b3e5dfaf18fb949ce8b8cR148) (at least, it worked in my quick testing). In other words, it didn't make sense to me why `purpose` would ever be missing, since it's present in the API response, and I think it's just getting dropped as part of some of the earlier transformations here, so a simpler option is to make sure that we preserve it, rather than filling it in from `value` if it's missing.

I didn't totally trace it through, but I suspect that making this change here will get pushed down to `map.js` via props as well; I removed the ternary from there and didn't notice any problems.

Reply:
Yes, thank you! This worked well on my end too and does makes more sense. I added the fix. 👍